modules/mwOpenLinkIdSystem.js

```diff
@@ -0,0 +1,179 @@
+  }
+  return true;
+}
+
+async function readCookie(name) {
+  name += '=';
+  for (var ca = document.cookie.split(/;\s*/), i = ca.length - 1; i >= 0; i--) {
+    if (!ca[i].indexOf(name)) {
+      return ca[i].replace(name, '');
+    }
+  }
+}
+
+async function writeCookie(name, value) {
+  var date = new Date();
+  date.setTime(date.getTime() + getExpirationDate()); // 3 year expiration
+  var expires = '; expires=' + date.toUTCString();
+
+  document.cookie = name + '=' + value + expires + '; path=/';
+}
+
+/* MW */
+
+async function generateUUID() { // Public Domain/MIT
+  var d = new Date().getTime();// Timestamp
+  var d2 = (performance && performance.now && (performance.now() * 1000)) || 0;// Time in microseconds since page-load or 0 if unsupported
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+    var r = Math.random() * 16;// random number between 0 and 16
+    if (d > 0) { // Use timestamp until depleted
+      r = (d + r) % 16 | 0;
+      d = Math.floor(d / 16);
+    } else { // Use microseconds since page-load if supported
+      r = (d2 + r) % 16 | 0;
+      d2 = Math.floor(d2 / 16);
+    }
+    return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16);
+  });
+}
+
+async function localStorageSupported() {
+  try {
+    var item = 1;
+    localStorage.setItem(item, item);
+    localStorage.removeItem(item);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+async function readLocalStorage(name) {
+  var localStorageItem = localStorage.getItem(name);
+  return (localStorageItem != 'undefined') ? localStorageItem : '';
+}
+
+async function confirmID() {
+  openLinkID.value = await readCookie(openLinkID.name) || '';
+  if (typeof openLinkID.value != 'undefined' &&
+        openLinkID.value !== null &&
+        openLinkID.value.length > 0) {
+  } else {
+    openLinkID.value = (await localStorageSupported() == true) ? await readLocalStorage(openLinkID.name) : openLinkID.value;
+  }
+  if (typeof openLinkID.value != 'undefined' &&
+        openLinkID.value !== null &&
+        openLinkID.value.length > 0) {
+  } else {
+    openLinkID.value = await generateUUID();
+    register(openLinkID.value);
+  }
+  return openLinkID.value;
+}
+
+async function register(olid) {
+  var accountId = (configParams.accountId != 'undefined') ? configParams.accountId : '';
+  var partnerId = (configParams.partnerId != 'undefined') ? configParams.partnerId : '';
+  var uid = (configParams.uid != 'undefined') ? configParams.uid : '';
+  var url = 'https://ol.mediawallahscript.com/?account_id=' + accountId +
+            '&partner_id=' + partnerId +
+            '&uid=' + uid +
+            '&olid=' + olid +
+            '&cb=' + Math.random()
+            ;
+  ajax(url);
+}
+
+async function setID() {
+  var olid = await confirmID();
+
+  if (localStorageSupported() == true) {
+    localStorage.setItem(openLinkID.name, olid);
+  }
+  writeCookie(openLinkID.name, olid);
+  return olid;
```

Comment:
Is it intentional to store the id value in local storage (if supported) and in the cookie?

Reply:
Seems like this submodule does store id based on the configparam.storage.

Reply:
there's no reason to read/write your own localstorage/cookies if you're using the UserId module to do so, as @SKOCHERI said

Reply:
Got it. Was keeping in sync with our standalone code...will make this compliant.